pandas/tseries/index.py

```diff
@@ -1581,10 +1590,14 @@ def insert(self, loc, item):
         """
 
         freq = None
+
+        if isinstance(item, np.datetime64):
+            # np.datetime64 doesn't have tz, once convert to Timestamp
+            item = Timestamp(item)
```

Comment:
hmm, this seems to be not necessary?

e.g. you are checking if its ``np.datetime64`` below so it will never be the case?